### DISCUSSION TITLE
The option to delete builds

### DISCUSSION THREAD

**Discussion by MostafaSolimanMO** [codemagic-ci-cd/codemagic-docs#2129] [Ideas]
### Use case

```Text
I want to be able to delete the failed builds or even the successful ones
```

### Idea

```Text
An option button for delete 🤷🏼‍♂️
```

**Comment by Ycoi** [comment]
Hi @MostafaSolimanMO! Thanks for contacting us. Is there some particular reason you want to be able to delete builds? Can you elaborate on that more?

**Reply by MostafaSolimanMO** [reply]
There are two main reasons that need to be addressed. First, we often share the build links, and occasionally, the publishing process fails. However, the build artifacts remain accessible. Therefore, we would like the ability to delete the build when it fails, ensuring that only successful builds retain access to the artifacts.

The second issue is more personal. Sometimes, build failures occur due to minor or trivial issues. In such cases, I would like to remove these unsuccessful 'red' builds from the build tab. having only successful builds like a 10x developer :)

**Reply by Ycoi** [reply]
Thanks for the clarification. May I ask how do you share the build links? If you use the email or Slack publishing of Codemagic, you can set it so that this notification is sent only for successful builds. Also note that the artifact link in email and Slack notification is only valid for 24 hours (this can be changed in Team settings > Artifact download links) so this way the builds won’t stay accessible forever.

Another way to share the build links is to use [Build Dashboards](https://docs.codemagic.io/yaml-publishing/build-dashboards/) That way you can for example create a Dashboard that only shows all the successful builds and your testers can always pick the latest one. I think this could also solve part of your second issue. I personally think that hiding development issues by deleting failed builds isn’t a good convention and it’s good to have a history of that. But of course Testers or other stakeholders who need access only to the good builds don’t necessarily need to see all the failed builds.

So I think better way would be to keep the complete build history for development, but then only deliver links to successful builds to Testers or other stakeholders using either by Build Dashboards or by only sending the Slack or Email notification to your testers / stakeholders if the build succeeds.

**Reply by Ycoi** [reply]
Thanks again for the idea! I’ll close this thread since ability to delete builds is not at least in our short term planning and we are focusing on developming and delivering other improvements. As said abvove I also feel that keeping build history is also important and your issues can most likely be solved using the adivece above.

If you have any questions about setting up Build Dashboards or encounter other issues, you can always open a thread in the Q&A section. And if you feel this feature still would be important for you or you have other ideas on how Codemagic can be improved, please go ahead and open new threads here on the Ideas section.
